types/spectest/generate/main.go

```diff
@@ -27,19 +28,40 @@ func main() {
 		}
 		all[n] = t
 	}
+	log.Printf("found %d tests\n", len(all))
+	if len(all) != len(spectest.AllTests) {
+		panic("did not generate all tests\n")
+	}
 
-	byts, err := json.Marshal(all)
-	if err != nil {
+	// write small json files for each test
+	// try to create directory if it doesn't exist
+	_, basedir, _, ok := runtime.Caller(0)
+	if !ok {
+		panic("no caller info")
+	}
+	testsDir := filepath.Join(strings.TrimSuffix(basedir, "main.go"), "tests")
+	if err := os.MkdirAll(testsDir, 0700); err != nil && !os.IsExist(err) {
```

Comment:
after all files are written do `os.Chmod(testsDir, 0444)`

This will make files read only... meaning CI that will use the files will always be able to read them